test/shape_attributes.cc

```diff
@@ -0,0 +1,108 @@
+               "max_route_distance_factor":3,"max_route_time_factor":3,"max_search_radius":100,"route":true,
+               "search_radius":50,"sigma_z":4.07,"turn_penalty_factor":200}},
+      "service_limits": {
+        "auto": {"max_distance": 5000000.0, "max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "auto_shorter": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "bicycle": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50},
+        "bus": {"max_distance": 5000000.0,"max_locations": 50,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "hov": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "taxi": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "isochrone": {"max_contours": 4,"max_distance": 25000.0,"max_locations": 1,"max_time": 120},
+        "max_avoid_locations": 50,"max_radius": 200,"max_reachability": 100,"max_alternates":2,
+        "multimodal": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 0.0,"max_matrix_locations": 0},
+        "pedestrian": {"max_distance": 250000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50,"max_transit_walking_distance": 10000,"min_transit_walking_distance": 1},
+        "skadi": {"max_shape": 750000,"min_resample": 10.0},
+        "trace": { "max_best_paths": 4, "max_best_paths_shape": 100, "max_distance": 200000.0, "max_gps_accuracy": 100.0, "max_search_radius": 100, "max_shape": 16000 },
+        "transit": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50},
+        "truck": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50}
+      }
+    })");
+
+  conf.get_child("mjolnir").put("tile_dir", VALHALLA_SOURCE_DIR "test/traffic_matcher_tiles");
+  return conf;
+}
+
+void test_attributes() {
+  auto conf = make_conf();
+  tyr::actor_t actor(conf);
+
+  actor.trace_attributes(R"({"shape":[{"lat":40.546115,"lon":-76.385076},
+      {"lat":40.544232,"lon":-76.385752}],"costing":"auto","shape_match":"walk_or_snap",
+      "filters":{"attributes":["shape_attributes.time","shape_attributes.length"],
+      "action":"include"}})");
+  actor.cleanup();
+  auto attributes_json = actor.trace_attributes(R"({"shape":[{"lat":40.546115,"lon":-76.385076},
+      {"lat":40.544232,"lon":-76.385752}],"costing":"auto","shape_match":"walk_or_snap",
+      "filters":{"attributes":["shape_attributes.time","shape_attributes.length"],
+      "action":"include"}})");
+  actor.cleanup();
+  auto attributes = json_to_pt(attributes_json);
+  attributes_json.find("length");
+  attributes_json.find("time");
```

Comment:
this test doesnt make sense to me. it calls trace attributes twice, the first time complete ignoring the result, could see maybe doing that in the general actor test to make sure that results dont bleed through or something but it doesnt make sense here.

additoinally this test doesnt test anything. it parses the response but then doesnt validate anything. you should make sure you have a multi edge route, you should also decode the shape and measure that the length between each is appx the same as the length in the shape attributes. you can then also ask for the edge time and validate that that makes sense as well.